util/export_mmcif.py

```diff
@@ -324,10 +324,12 @@ def write(self, experiments, reflections):
         self._cif["dials"] = cif_block
 
         # Print to file
-        if filename.endswith(".bz2"):
-            open_fn = bz2.open
-        elif filename.endswith(".gz"):
+        if self.params.mmcif.compress is not None:
+            filename += "." + self.params.mmcif.compress
+        if self.params.mmcif.compress == "gz":
             open_fn = gzip.open
+        elif self.params.mmcif.compress == "bz2":
+            open_fn = bz2.open
         else:
             open_fn = open
         with open_fn(filename, "wt") as fh:
```

Comment:
```suggestion
        with open_fn(filename, "w") as fh:
```
`t` just raises questions and is the default anyway

Reply:
Unfortunately this fails abruptly with the message `a bytes-like object is required, not 'str'`

Reply:
no stacktrace and I've not dug into where this happens, but suspect it is within `iotbx/cif/`

Reply:
Yes, it is here: https://github.com/cctbx/cctbx_project/blob/8737e695766c1c7395da6650e5a89777a2d3f145/iotbx/cif/model.py#L83

Reply:
did I ever mention that this print-to-file pattern is so, so, so special...

Leave it then :slightly_smiling_face: